### PR TITLE
Added .NET Core compatibility 

### PR DESCRIPTION
**Pull request by tom-corwin**

### CODE REVIEW

**Comment by NicolasDorier** [comment]
please, fix the indent that cause unecessary diff.

**Comment by tom-corwin** [comment]
I fixed most of the diff, but the rest I left so every .cs file is uniform to the tabs not spaces rules.

**Comment by NicolasDorier** [comment]
Remove the changes to the JSON file, and squash your commits please.

File: NBitcoin/Utils.cs [hunk 1]
```diff
@@ -10,6 +10,7 @@
 using System.Threading;
 using NBitcoin.Protocol;
 using System.Runtime.ExceptionServices;
+using System.Threading.Tasks;
```
Comment (NicolasDorier) [inline NBitcoin/Utils.cs]: do you need it ?

**Comment by tom-corwin** [comment]
I'm sorry this has been a hassle, but I'm not sure how to squash the commits. And what changes to the `json` file are you referring to?

**Comment by NicolasDorier** [comment]
http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Comment by tom-corwin** [comment]
I'm running the following command:
```
git rebase -i HEAD~9
```
And after I edit the file that comes up, I get the error:
```
error: could not apply cd9aa0d... Add NETCore symbols and project files

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".
Could not apply cd9aa0d72a86963d4b2ac44c45f92b54c7d02b02... Add NETCore symbols and project files
```
Any ideas? I'm not familiar with rebase and haven't sound a fix for it.

**Comment by NicolasDorier** [comment]
I fixed the indent and squashed on b591359ea3ff7ed3b9d439d1dadb7ddfd8eb8abe. Please, learn how to make squashes and be careful about breaking the formatting.

**Comment by tom-corwin** [comment]
Sorry about that. I guess I bit off more than I could chew in some aspects of this.

**Pull request closed without merging**